compiler_opt/distributed/worker.py

```diff
@@ -30,24 +30,23 @@ def is_priority_method(cls, method_name: str) -> bool:
 
 
 # Dask's Futures are limited. This captures that.
-class WorkerFuture(Generic[T], metaclass=abc.ABCMeta):
+class WorkerFuture(Protocol[T]):
 
-  @abc.abstractmethod
   def result(self) -> T:
     raise NotImplementedError()
 
-  @abc.abstractmethod
   def done(self) -> bool:
     raise NotImplementedError()
 
+  def add_done_callback(self, fn) -> None:
+    raise NotImplementedError
+
 
 def wait_for(futures: Iterable[WorkerFuture]):
   """Dask futures don't support more than result() and done()."""
   for f in futures:
-    try:
-      _ = f.result()
-    except:  # pylint: disable=bare-except
-      pass
+    while not f.done():
+      time.sleep(0.1)
```

Comment:
could the dummy future be an actual concurrency.future.Future that resolves when the worker's future does?